### PR TITLE
initial m-06 assessment

### PR DESCRIPTION
**Pull request by ctobler40**

### CODE REVIEW

File: week-06/assessment/field-agent/src/main/java/learn/field_agent/controllers/SecurityClearanceController.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    // Only allow deletion if a security clearance key isn't referenced.
+    @DeleteMapping("/{securityClearanceId}")
+    public ResponseEntity<Void> deleteById(@PathVariable int securityClearanceId)
+    {
+        if (securityClearanceService.deleteById(securityClearanceId))
+        {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
```
Comment (esin87) [inline week-06/assessment/field-agent/src/main/java/learn/field_agent/controllers/SecurityClearanceController.java:79]: You have some great logic in the service that returns an informative result, but you're not using it here. How could you update this to return result information instead of status codes? What status code would you return if the user tried to delete a SC in use?

File: week-06/assessment/field-agent/src/main/java/learn/field_agent/controllers/GlobalErrorHandler.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//import org.springframework.http.converter.HttpMessageNotReadableException;
+//import org.springframework.web.HttpMediaTypeNotSupportedException;
+//import org.springframework.web.bind.annotation.ControllerAdvice;
+//import org.springframework.web.bind.annotation.ExceptionHandler;
+//
+//import java.sql.SQLSyntaxErrorException;
+//
+//// Use the @ControllerAdvice annotation to register an exception handler for all controllers.
+//@ControllerAdvice
+//public class GlobalErrorHandler
```
Comment (esin87) [inline week-06/assessment/field-agent/src/main/java/learn/field_agent/controllers/GlobalErrorHandler.java:14]: Let's get this working.

File: week-06/assessment/field-agent/src/main/java/learn/field_agent/data/MissionJdbcTemplateRepository.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                "a.agency_id, a.short_name, a.long_name, " +
+                "m.mission_id, m.code_name, m.notes, m.start_date, m.projected_end_date, m.actual_end_date, m.operational_cost, m.agency_id " +
+                "from agency a " +
+                "inner join mission m on a.agency_id = m.agency_id;";
+
+        return jdbcTemplate.query(sql, new MissionMapper());
+    }
+
+    @Override
+    @Transactional
```
Comment (esin87) [inline week-06/assessment/field-agent/src/main/java/learn/field_agent/data/MissionJdbcTemplateRepository.java:39]: We only need transactional if we want to group multiple queries/operations as one SQL _transaction_, so that they wholly fail or wholly succeed.

**Review by esin87 — CHANGES_REQUESTED** [review]
Really nice job here, CJ! I'm impressed you got so much done on the stretch goals. Code is clean, easy to read, and idiomatic. This project does a nice job illustrating everything we've been learning over the past few weeks. Here are the changes I'd like you to make: 

- [ ] Add tests for SC data and service classes (at a minimum, feel free to include tests for missions and aliases) 
- [ ] Implement global error handling 
- [ ] Return a more detailed response body on the SC delete method on the controller 

Grade: Not Yet
